### ISSUE TITLE
label size

### ISSUE THREAD

**Issue by STITCHESHHH** [ultralytics/ultralytics#9423]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Thanks to the code you guys gave me, I got the trajectory tracking done and plotted the trajectory line!
However, in the result video, the label size and confidence box is too big and it is blocked, how can I change the label size and so on in the result video?
![image](https://github.com/ultralytics/ultralytics/assets/147716690/1eaf5bdd-f265-4c8f-be9a-344da12f720e)
Looking forward to your help!

**Comment by glenn-jocher** [comment]
@STITCHESHHH hello! 👋 Great to hear you got the trajectory tracking done successfully!

For adjusting the label size and the confidence box in the result video, you can modify the `line_width` and `font_size` parameters in the `plot()` method. Here's a quick example to help you fine-tune those visuals:

```python
results = model('your_video.mp4')
results[0].plot(line_width=2, font_size=10)  # Adjust these values as needed
results[0].show()  # To display
results[0].save('path/to/save/your_modified_video.mp4')  # To save the modified video
```

Adjust the `line_width` for the border thickness of boxes and `font_size` for the size of the text, including labels and confidence scores. Tweak these values to find the best fit for your video.

Hope this helps, and if you have any more questions, feel free to ask! 🚀

**Comment by STITCHESHHH** [comment]
Thank you for your helps!

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH hello! 👋 Great to hear you got the trajectory tracking done successfully!
> 
> For adjusting the label size and the confidence box in the result video, you can modify the `line_width` and `font_size` parameters in the `plot()` method. Here's a quick example to help you fine-tune those visuals:
> 
> ```python
> results = model('your_video.mp4')
> results[0].plot(line_width=2, font_size=10)  # Adjust these values as needed
> results[0].show()  # To display
> results[0].save('path/to/save/your_modified_video.mp4')  # To save the modified video
> ```
> 
> Adjust the `line_width` for the border thickness of boxes and `font_size` for the size of the text, including labels and confidence scores. Tweak these values to find the best fit for your video.
> 
> Hope this helps, and if you have any more questions, feel free to ask! 🚀

Thank you for your helps!

**Comment by glenn-jocher** [comment]
@STITCHESHHH you're welcome! 😊 If you hit any bumps or have more questions down the road, don't hesitate to reach out. Happy coding! 🚀

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH you're welcome! 😊 If you hit any bumps or have more questions down the road, don't hesitate to reach out. Happy coding! 🚀

Thanks to your previous help, and thank you for the official model!I managed to make the tags etc. smaller in the resulting video. In this tracking, I used the official weights you provided as a detector. The result graph is below:
![image](https://github.com/ultralytics/ultralytics/assets/147716690/6724c296-4e92-47da-87fd-d608ed88fa5e)
But when I use a custom model it doesn't work very well, what improvements should I make during training to give my model higher detection accuracy? Customize the model results:
![image](https://github.com/ultralytics/ultralytics/assets/147716690/7f7ece7f-2603-47e9-92ec-13900dbccf42)
Supplement:
![image](https://github.com/ultralytics/ultralytics/assets/147716690/982dead3-7717-4dc8-b12f-f9eea9614068)
 they both have high mAP values
The training log is recorded as follows
![image](https://github.com/ultralytics/ultralytics/assets/147716690/fb55095b-9d01-49e8-a217-de6f598c80db)
![image](https://github.com/ultralytics/ultralytics/assets/147716690/23c144f6-b897-4e45-b667-ce76d8c29048)
Looking forward to your help! 😊

**Comment by glenn-jocher** [comment]
@STITCHESHHH, glad to hear you've made progress with the official model! 🌟 For improving a custom model, especially when it's not performing as well as expected despite high mAP values, consider the following tips:

1. **Data Quality:** Ensure your training dataset is well-labeled and diverse, covering different variations of the objects you're interested in detecting.

2. **Augmentation:** Increase the variety of your training data with more aggressive data augmentation. This could help the model generalize better.

3. **Hyperparameters:** Experiment with different hyperparameters. Sometimes, slightly adjusting learning rates, weight decay, or other parameters can lead to better model convergence.

4. **Model Architecture:** If your custom model is significantly different from the official ones, ensure it's well-suited for the task at hand. Sometimes, simpler models can outperform more complex ones on specific datasets.

5. **Batch Size and Learning Rate:** A larger batch size with a correspondingly adjusted learning rate can sometimes help the model learn more effectively.

Here's a quick example of how you might adjust data augmentation settings in your training configuration:

```yaml
augment:
  hue: 0.1  # Adjust hue
  saturation: 1.5  # Increase saturation for more color diversity
  exposure: 1.5  # Adjust exposure
  mosaic: True  # Enable mosaic augmentation
```

Remember, machine learning is iterative and experimental. Keep experimenting with different settings, and don't hesitate to dive deeper into the visualizations of where your model might be getting confused. Good luck, and reach out anytime! 😊

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH, glad to hear you've made progress with the official model! 🌟 For improving a custom model, especially when it's not performing as well as expected despite high mAP values, consider the following tips:
> 
> 1. **Data Quality:** Ensure your training dataset is well-labeled and diverse, covering different variations of the objects you're interested in detecting.
> 2. **Augmentation:** Increase the variety of your training data with more aggressive data augmentation. This could help the model generalize better.
> 3. **Hyperparameters:** Experiment with different hyperparameters. Sometimes, slightly adjusting learning rates, weight decay, or other parameters can lead to better model convergence.
> 4. **Model Architecture:** If your custom model is significantly different from the official ones, ensure it's well-suited for the task at hand. Sometimes, simpler models can outperform more complex ones on specific datasets.
> 5. **Batch Size and Learning Rate:** A larger batch size with a correspondingly adjusted learning rate can sometimes help the model learn more effectively.
> 
> Here's a quick example of how you might adjust data augmentation settings in your training configuration:
> 
> ```yaml
> augment:
>   hue: 0.1  # Adjust hue
>   saturation: 1.5  # Increase saturation for more color diversity
>   exposure: 1.5  # Adjust exposure
>   mosaic: True  # Enable mosaic augmentation
> ```
> 
> Remember, machine learning is iterative and experimental. Keep experimenting with different settings, and don't hesitate to dive deeper into the visualizations of where your model might be getting confused. Good luck, and reach out anytime! 😊
Well,thank you for your help,I will try to it!😊

**Comment by glenn-jocher** [comment]
@STITCHESHHH You're welcome! 😊 If you have any more questions or need further assistance along the way, feel free to ask. The journey of improving your model can be both challenging and rewarding. Keep experimenting and happy coding! 🚀

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH You're welcome! 😊 If you have any more questions or need further assistance along the way, feel free to ask. The journey of improving your model can be both challenging and rewarding. Keep experimenting and happy coding! 🚀

Hey!
I have some questions about video tracking. Is it possible that the tracking is done by a detector for feature extraction and a tracker for target tracking?
When tracking a video, does the detector split the video into pictures or does the tracker split the video into pictures to track the target frame by frame?
While tracking a video, can I change the size of the recognized image from its original size?
Sincerely looking forward to your help!😀

**Comment by glenn-jocher** [comment]
Hey @STITCHESHHH! 👋

Absolutely, the tracking process often involves a detector for identifying objects and a tracker that maintains object identity across frames. Typically, the detector runs first to identify objects in each frame. The tracker then uses these detections to maintain consistency of object identities over time.

For video tracking, the video doesn't get split into pictures explicitly. Instead, the video is processed frame by frame, effectively treating each frame as an individual picture for detection and tracking.

Yes, you can change the size of the processed frames. When initializing your model for inference, you can specify the `imgsz` parameter to set the desired input size, like so:
```python
model = YOLO('yolov8n.pt', imgsz=640)  # Example: changes input size to 640x640
```
This resizes the frames for processing, which can impact both performance and detection accuracy.

Hope this clears things up! If you've got more questions, I'm here to help. 😄

**Comment by glenn-jocher** [comment]
@STITCHESHHH hey there! 😄

You're right, high-resolution videos like 3840×2160 can be challenging for processing. You've already taken a great step by setting `imgsz=640`, which rescales frames to a more manageable resolution.

If you're looking to further customize frame processing, consider adjusting the `imgsz` based on your needs or computational resources. For example, if you're dealing with very high-res videos and notice performance issues, you might reduce `imgsz` even further. Just keep in mind that smaller sizes might affect detection accuracy.

Here's how you can adjust it:
```python
model = YOLO('your_model_path.pt', imgsz=320)  # Adjust imgsz as needed
```

This adjustment is a balance: smaller sizes increase speed but might miss finer details, while larger sizes catch more details at the cost of processing speed.

Let me know if this helps or if you have any more questions!

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH hey there! 😄
> 
> You're right, high-resolution videos like 3840×2160 can be challenging for processing. You've already taken a great step by setting `imgsz=640`, which rescales frames to a more manageable resolution.
> 
> If you're looking to further customize frame processing, consider adjusting the `imgsz` based on your needs or computational resources. For example, if you're dealing with very high-res videos and notice performance issues, you might reduce `imgsz` even further. Just keep in mind that smaller sizes might affect detection accuracy.
> 
> Here's how you can adjust it:
> 
> ```python
> model = YOLO('your_model_path.pt', imgsz=320)  # Adjust imgsz as needed
> ```
> 
> This adjustment is a balance: smaller sizes increase speed but might miss finer details, while larger sizes catch more details at the cost of processing speed.
> 
> Let me know if this helps or if you have any more questions!

@glenn-jocher Hey, thank you for your reply!
When I was doing the tracking, based on the method you suggested, in the 
model = YOLO('yolov8n.pt', imgsz=640)
 
 with imgsz=640 added inside.
However, an error alert appeared
![image](https://github.com/ultralytics/ultralytics/assets/147716690/e765a685-9e64-4cc9-a38f-25d93f190aec)
What can I do to solve this problem?
Sincerely looking forward to your reply!😄

**Comment by glenn-jocher** [comment]
Hey @STITCHESHHH! 😄

It looks like you encountered a small hiccup there. Would you mind trying to set `imgsz` during inference like this instead:

```python
model = YOLO('yolov8n.pt')  # Loading the model
results = model.predict(source='path/to/video.mp4', imgsz=640)  # Setting imgsz during prediction
```

This approach sets the image size at the time of prediction, which should align with how the YOLO class handles parameters. Give it a shot and let us know if it gets you past that error! 🚀

Looking forward to hearing back from you!

**Comment by STITCHESHHH** [comment]
> Hey @STITCHESHHH! 😄
> 
> It looks like you encountered a small hiccup there. Would you mind trying to set `imgsz` during inference like this instead:
> 
> ```python
> model = YOLO('yolov8n.pt')  # Loading the model
> results = model.predict(source='path/to/video.mp4', imgsz=640)  # Setting imgsz during prediction
> ```
> 
> This approach sets the image size at the time of prediction, which should align with how the YOLO class handles parameters. Give it a shot and let us know if it gets you past that error! 🚀
> 
> Looking forward to hearing back from you!

Thanks for the suggestion😄, I've tried it and it still doesn't seem to run successfully.

**Comment by hkhkhkhkhkhkhkhk** [comment]
hey there! 😄  I wonder if the image size must be 640*640 when training my dataset? I tried other image sizes, but it didn't work.

**Comment by STITCHESHHH** [comment]
> Hey @STITCHESHHH! 😄
> 
> It looks like you encountered a small hiccup there. Would you mind trying to set `imgsz` during inference like this instead:
> 
> ```python
> model = YOLO('yolov8n.pt')  # Loading the model
> results = model.predict(source='path/to/video.mp4', imgsz=640)  # Setting imgsz during prediction
> ```
> 
> This approach sets the image size at the time of prediction, which should align with how the YOLO class handles parameters. Give it a shot and let us know if it gets you past that error! 🚀
> 
> Looking forward to hearing back from you!

@glenn-jocher Hey😄
I thought of a question about this image resolution, if I use a larger dataset resolution instead of 640×640 when training the dataset, will the generated custom model work better when doing object tracking
If yes, then what is the maximum resolution that can be supported during training
Sincerely look forward to your reply!

**Comment by glenn-jocher** [comment]
Hey @STITCHESHHH! 😄

Absolutely, using higher resolution images for training can potentially improve your model's ability to detect smaller features and nuances, leading to better tracking performance. However, the key is finding a balance between image size, model complexity, and available computational resources. Larger images require more memory and processing power.

While there isn't a strict "maximum resolution," practical limits are often dictated by your hardware capabilities. Many users find a sweet spot around 640x640 to 1024x1024 for training, but if your setup allows, you can experiment with higher resolutions. Just ensure that you adjust `imgsz` during training to match:

```python
yolo train data=your_dataset.yaml imgsz=1024  # Example for a higher resolution
```

Remember, experimentation is key! Adjust, try out different settings, and observe how your model performs. 😊

**Comment by STITCHESHHH** [comment]
Hey @glenn-jocher 😄This image is a screenshot of the runtime process when doing video tracking, where 576×1024 is not the resolution size being detected for each frame of the image? Can I change it to 1024×1024?
![image](https://github.com/ultralytics/ultralytics/assets/147716690/f77972c8-0460-4ee1-80be-7ba55051aa7a)

**Comment by glenn-jocher** [comment]
@STITCHESHHH hey there! 😄 Indeed, the 576×1024 you're seeing is the current resolution being processed. To change to 1024×1024, you just need to adjust the `imgsz` parameter in your tracking command. For instance:

```python
results = model.track(source='path/to/video.mp4', imgsz=1024)
```

This will process your video frames at 1024×1024 pixels. Keep in mind this might increase memory and computational requirements. Happy tracking!

**Comment by STITCHESHHH** [comment]
> @STITCHESHHH hey there! 😄 Indeed, the 576×1024 you're seeing is the current resolution being processed. To change to 1024×1024, you just need to adjust the `imgsz` parameter in your tracking command. For instance:
> 
> ```python
> results = model.track(source='path/to/video.mp4', imgsz=1024)
> ```
> 
> This will process your video frames at 1024×1024 pixels. Keep in mind this might increase memory and computational requirements. Happy tracking!

Hey@glenn-jocher 😄 I have 5 categories in my dataset, including motorcycle, but there is no result for motorcycle at the end of training, can you help me?
![image](https://github.com/ultralytics/ultralytics/assets/147716690/b4bca531-724a-4528-882b-96bbe382645b)
![image](https://github.com/ultralytics/ultralytics/assets/147716690/9e1f59a6-b6fc-4a97-a72c-140ae63c7de3)

I am looking forward to your reply！

**Comment by glenn-jocher** [comment]
Hey @STITCHESHHH 😊! It looks like motorcycles might not be detected due to several reasons such as insufficient or imbalanced training data for that category, or the model might require fine-tuning for better detection. To address this, ensure your dataset has a good amount of diverse and representative images of motorcycles. Also, tweaking the training parameters might help. For instance, experimenting with different learning rates or more training epochs could improve detection. You could try something like this for retraining:

```python
yolo train data=coco128.yaml model=yolov8n.pt epochs=100 lr0=0.01 imgsz=1024
```

Replace `coco128.yaml` with your dataset configuration file, and adjust the `epochs`, `lr0`, and `imgsz` as needed. Keep us posted on your progress! 🚀

**Comment by STITCHESHHH** [comment]
> Hey @STITCHESHHH 😊! It looks like motorcycles might not be detected due to several reasons such as insufficient or imbalanced training data for that category, or the model might require fine-tuning for better detection. To address this, ensure your dataset has a good amount of diverse and representative images of motorcycles. Also, tweaking the training parameters might help. For instance, experimenting with different learning rates or more training epochs could improve detection. You could try something like this for retraining:
> 
> ```python
> yolo train data=coco128.yaml model=yolov8n.pt epochs=100 lr0=0.01 imgsz=1024
> ```
> 
> Replace `coco128.yaml` with your dataset configuration file, and adjust the `epochs`, `lr0`, and `imgsz` as needed. Keep us posted on your progress! 🚀

Thanks for the suggestion😄,I have got it!

**Comment by glenn-jocher** [comment]
Great to hear that, @STITCHESHHH! 😄 If you have any more questions or run into any issues as you progress, feel free to reach out. Happy training, and we're excited to see your model improve! 🚀
